internal/pkg/v2/README.md

```diff
@@ -0,0 +1,296 @@
+## v2 Package
+
+[TOC]
+
+### Overview
+
+This package encapsulates the APIv2 implementation.
+
+The implementation is based on [the presentation](https://zoom.us/rec/share/98x2E4vXxEFLZtbt7gLHV4oCEafMaaa803VP-vQFmk9FjGQVZvvBa-EniNhqLmEt) ([slides](https://wiki.edgexfoundry.org/download/attachments/329472/APIV2_20200123.pdf?version=1&modificationDate=1579821274000&api=v2)) given to the Core Working Group on January 23, 2020.
+
+#### Disclaimer
+
+This document and the implementation it describes is a work in progress and subject to change.  The architectural approach being taken is evolutionary; the existing implementation does not currently support all aspects of the final design.
+
+
+
+### Approach
+
+The scope of APIv2 prevents complete implementation across all services within a single EdgeX release cycle.  The community has decided to release Geneva with APIv1 and "experimental/beta" APIv2 support for  some subset of services.  The intention is for the future EdgeX 2.0 release to include only APIv2 endpoints.
+
+APIv2 is being treated as greenfield development.  This reduces risk by minimizing overlap with APIv1 implementation.  It ensures APIv2 is able to be developed and evolve separately from APIv1.  It also provides a clear delineation between APIv1 and APIv2 and makes it easier (and less likely to introduce last-minute issues) to remove the APIv1 functionality for the future EdgeX 2.0 release.
+
+To this end, the community decided that APIv2 persistence will be separate from APIv1 persistence.  This supports independent schema development at the expense of data access across the two API versions.  That is, data persisted via APIv1 will not be available via APIv2 (and vice-versa).
```

Comment:
I guess I missed this discussion. While I understand why it was made, I will point out that as all the current services use the V1 APIs, we can't really do any informal testing using v2 APIs to populate configuration, metadata, or runtime objects and then consume/validate them with v1 endpoints. Did we consider some sort of configuration flag to enable v2 endpoints and keep them using the same persistence back-end?

Reply:
Decision was made in a recent working group call (I frankly can't remember which one at the moment) to keep v1 and v2 completely separate down to and including persistence.  No overlap whatsoever.  This enables us to evolve the v2 persistence schema independently of v1.    It also minimizes the risk of v2 implementation polluting v1 (and vice-versa) and the risk of impacting the v2 implementation when we eventually remove the v1 implementation.

Reply:
FYI -- Decision was in Core WG 30-Jan.